durable-storage/src/persistence_layer.rs

```diff
@@ -225,24 +229,61 @@ impl<'a> PersistenceLayer<'a> {
 
         self.checkpoint_at(&checkpoint_path)
     }
+}
+
+// Interface used by the Merkle layer which operates over implicitly hashed values.
+impl PersistenceLayer<'_> {
+    fn blob_cf(&self) -> &rocksdb::ColumnFamily {
+        self.db_instance
+            .cf_handle(BLOB_CF)
+            .expect("the rocksdb instance should always contain the data cf")
+    }
 
+    /// Retrieves the hashed data associated with its hash as the key.
+    pub fn blob_get(&self, key: &Hash) -> Result<impl AsRef<[u8]>, Error> {
+        self.db_instance
+            .get_pinned_cf(self.blob_cf(), key.as_ref())?
+            .ok_or(Error::KeyNotFound)
+    }
+
+    /// Sets a value for the given key.
+    pub fn blob_set(&mut self, blob: &HashedData) -> Result<(), Error> {
+        self.make_temporary()?;
+
+        Ok(self
+            .db_instance
+            .put_cf(self.blob_cf(), blob.hash, blob.value)?)
+    }
+
+    /// Deletes a value associated with the given key.
+    pub fn blob_delete(&mut self, key: &Hash) -> Result<(), Error> {
+        self.make_temporary()?;
+
+        Ok(self.db_instance.delete_cf(self.blob_cf(), key.as_ref())?)
+    }
+}
+
+// Interface used by the Data layer which operates over raw key-value pairs.
+impl PersistenceLayer<'_> {
     /// Retrieves a value associated with the given key.
-    pub fn get(&self, key: &Hash) -> Result<impl AsRef<[u8]>, Error> {
-        self.db_instance.get_pinned(key)?.ok_or(Error::KeyNotFound)
+    pub fn get(&self, key: impl AsRef<[u8]>) -> Result<impl AsRef<[u8]>, Error> {
+        self.db_instance
+            .get_pinned(key.as_ref())?
+            .ok_or(Error::KeyNotFound)
     }
 
     /// Sets a value for the given key.
-    pub fn set(&mut self, blob: &HashedData) -> Result<(), Error> {
+    pub fn set(&mut self, key: impl AsRef<[u8]>, value: &impl AsRef<[u8]>) -> Result<(), Error> {
```

Comment:
```suggestion
    pub fn set(&mut self, key: impl AsRef<[u8]>, value: impl AsRef<[u8]>) -> Result<(), Error> {
```

Resolve when addressed.